sound/soc/intel/boards/sof_sdw_max98373.c

```diff
@@ -68,7 +68,19 @@ int sof_sdw_mx8373_init(const struct snd_soc_acpi_link_adr *link,
 	if (info->amp_num == 2)
 		dai_links->init = spk_init;
 
+	info->late_probe = true;
```

Comment:
do I understand it correctly, that you're modifying here a global static common struct? Are we sure no other driver will want to use it?

Reply:
we also increase info->amp_num in this function.   It is only used by sof_sdw machine driver for it is a static struct define in sof_sdw.c

Reply:
So what happens if you have several codecs of the same type?

Reply:
@lyakh can you clarify your question? we do have configurations with 2 amplifiers of the same type, not sure what you see as wrong or incorrect?

Reply:
@plbossart @RanderWang I think I'm getting it slowly. `codec_info_list` is a global array of objects - one per device type. Each of these objects is shared by all instances of that respective device - both for reading and writing. That's also how `.amp_num` is used. And there's no locking because (so far) those fields are changed from `.init()` and it's called for each device instance successively so no race is possible. This seems to be ok so far, but is a bit... strange. Can we also have codecs of different types mixed on a system? Would we then have to count them separately or together?

Reply:
yes, no race for sdw-sof is executed serially.   We have one case on TGL:  1308 i2s + 711 sdw + HDMI. 

-